src/server/save.rs

```diff
@@ -388,11 +388,57 @@ impl SavedFile {
     }
 }
 
+/// Wrapper for the fields of an `Entries`. Stores text fields in a `Vec<(String, String)>` to
+/// enable accessing fields with duplicated names.
+
+#[derive(Debug, Default)]
+pub struct FieldsWrapper(Vec<(String, String)>);
+
+impl FieldsWrapper {
+
+    /// Converts the contents into a multivalued hashmap
+    pub fn into_multivalued_map(self) -> HashMap<String, Vec<String>> {
+        let mut out = HashMap::new();
+        for (key, val) in self.0.into_iter() {
+            out.entry(key).or_insert_with(Vec::new).push(val);
+        }
+        out
+    }
+
+    /// Converts the contents into a single-valued hashmap
+    pub fn into_map(self) -> HashMap<String, String> {
```

Comment:
This seems to be strictly less useful than `into_multivalued_map`

Reply:
I've included this in case anyone was relying on the prior behavior of dropping values for duplicated field names, or if someone knows they won't have duplicate fields and wants the ease of working with a `HashMap<String, String>`.